src/frequenz/sdk/_internal/asyncio.py

```diff
@@ -18,6 +18,8 @@ async def cancel_and_await(task: asyncio.Task[Any]) -> None:
     Args:
         task: The task to be cancelled and waited for.
     """
+    if task.done():
+        return
```

Comment:
Also maybe put this behavior in the docstring too.

Reply:
done